lib/Dialect/Comb/CombFolds.cpp

```diff
@@ -3291,3 +3291,90 @@ LogicalResult ICmpOp::canonicalize(ICmpOp op, PatternRewriter &rewriter) {
 
   return failure();
 }
+
+//===----------------------------------------------------------------------===//
+// TruthTableOp
+//===----------------------------------------------------------------------===//
+
+// Canonicalize truth tables by folding to constant when output does not depend
+// on any of the inputs or simplifying when the output depends on only a single
+// input.
+LogicalResult TruthTableOp::canonicalize(TruthTableOp op,
+                                         PatternRewriter &rewriter) {
+  if (isOpTriviallyRecursive(op))
+    return failure();
+
+  const auto inputs = op.getInputs();
+  const auto table = op.getLookupTable();
+  size_t numInputs = inputs.size();
+  size_t tableSize = table.size();
+
+  if (numInputs == 0 || tableSize == 0 || tableSize != (1ull << numInputs))
+    return failure();
+
+  // Check if the table can be folded to just a constant.
+  bool firstValue = cast<BoolAttr>(table[0]).getValue();
+  bool allSame = llvm::all_of(table, [firstValue](Attribute attr) {
+    return cast<BoolAttr>(attr).getValue() == firstValue;
+  });
```

Comment:
nit: Probably this works (comparing attributes). 
```suggestion
  bool allSame = llvm::all_equal(table);
```